common/components/WalletDecrypt/WalletDecrypt.tsx

```diff
@@ -133,311 +135,319 @@ const SECURE_WALLETS = Object.values(SecureWalletName);
 const INSECURE_WALLETS = Object.values(InsecureWalletName);
 const MISC_WALLETS = Object.values(MiscWalletName);
 
-export class WalletDecrypt extends Component<Props, State> {
-  // https://github.com/Microsoft/TypeScript/issues/13042
-  // index signature should become [key: Wallets] (from config) once typescript bug is fixed
-  public WALLETS: Wallets = {
-    [SecureWalletName.WEB3]: {
-      lid: WEB3_TYPE && WEB3_TYPES[WEB3_TYPE] ? WEB3_TYPES[WEB3_TYPE].lid : 'x_Web3',
-      icon: WEB3_TYPE && WEB3_TYPES[WEB3_TYPE] && WEB3_TYPES[WEB3_TYPE].icon,
-      description: 'ADD_Web3Desc',
-      component: Web3Decrypt,
-      initialParams: {},
-      unlock: this.props.unlockWeb3,
-      attemptUnlock: true,
-      helpLink: `${knowledgeBaseURL}/migration/moving-from-private-key-to-metamask`
-    },
-    [SecureWalletName.LEDGER_NANO_S]: {
-      lid: 'x_Ledger',
-      icon: LedgerIcon,
-      description: 'ADD_HardwareDesc',
-      component: LedgerNanoSDecrypt,
-      initialParams: {},
-      unlock: this.props.setWallet,
-      helpLink: 'https://support.ledgerwallet.com/hc/en-us/articles/115005200009'
-    },
-    [SecureWalletName.TREZOR]: {
-      lid: 'x_Trezor',
-      icon: TrezorIcon,
-      description: 'ADD_HardwareDesc',
-      component: TrezorDecrypt,
-      initialParams: {},
-      unlock: this.props.setWallet,
-      helpLink: 'https://doc.satoshilabs.com/trezor-apps/mew.html'
-    },
-    [InsecureWalletName.KEYSTORE_FILE]: {
-      lid: 'x_Keystore2',
-      example: 'UTC--2017-12-15T17-35-22.547Z--6be6e49e82425a5aa56396db03512f2cc10e95e8',
-      component: KeystoreDecrypt,
-      initialParams: {
-        file: '',
-        password: ''
+const WalletDecrypt = withRouter<Props>(
```

Comment:
You can spare yourself the indentation by moving the withRouter call to the export below, ala connect.

Reply:
I tried that but I wasn't able to get the types to work properly, not sure why :/ I'll have another look and see if I can clean it up